rules/msk_topic_config_comments.go

```diff
@@ -0,0 +1,317 @@
+package rules
+
+import (
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/logger"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+// MSKTopicConfigCommentsRule checks comments on time and bytes values.
+type MSKTopicConfigCommentsRule struct {
+	tflint.DefaultRule
+}
+
+func (r *MSKTopicConfigCommentsRule) Name() string {
+	return "msk_topic_config_comments"
+}
+
+func (r *MSKTopicConfigCommentsRule) Enabled() bool {
+	return true
+}
+
+func (r *MSKTopicConfigCommentsRule) Link() string {
+	return ReferenceLink(r.Name())
+}
+
+func (r *MSKTopicConfigCommentsRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (r *MSKTopicConfigCommentsRule) Check(runner tflint.Runner) error {
+	isRoot, err := isRootModule(runner)
+	if err != nil {
+		return err
+	}
+	if !isRoot {
+		logger.Debug("skipping child module")
+		return nil
+	}
+
+	resourceContents, err := runner.GetResourceContent(
+		"kafka_topic",
+		&hclext.BodySchema{
+			Attributes: []hclext.AttributeSchema{
+				{Name: "config"},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("getting kafka_topic contents: %w", err)
+	}
+
+	for _, topicResource := range resourceContents.Blocks {
+		if err := r.validateTopicConfig(runner, topicResource); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *MSKTopicConfigCommentsRule) validateTopicConfig(runner tflint.Runner, topic *hclext.Block) error {
```

Comment:
Naming: this isn't validating the config

```suggestion
func (r *MSKTopicConfigCommentsRule) validateComments(runner tflint.Runner, topic *hclext.Block) error {
```

Reply:
Good point 👍